pulsar-broker/src/test/java/org/apache/pulsar/compaction/CompactorTest.java

```diff
@@ -177,6 +184,54 @@ public void testCompaction() throws Exception {
         compactAndVerify(topic, expected, true);
     }
 
+    @Test
+    public void testAllCompactedOut() throws Exception {
+        String topicName = "persistent://my-property/use/my-ns/testAllCompactedOut";
+        // set retain null key to true
+        TwoPhaseCompactor compactor = (TwoPhaseCompactor) ((PulsarCompactionServiceFactory)
+                pulsar.getCompactionServiceFactory()).getCompactor();
+        compactor.setTopicCompactionRetainNullKey(true);
+
```

Comment:
We can change config `topicCompactionRetainNullKey=true` and restart broker.

Reply:
good idea.
fixed, PTAL.